source/common/network/address_impl.cc

```diff
@@ -8,30 +8,86 @@
 namespace Network {
 namespace Address {
 
+Address::InstanceConstSharedPtr addressFromSockAddr(const sockaddr_storage& ss, socklen_t ss_len) {
+  if (ss_len != 0 && ss_len <= sizeof(sa_family_t)) {
```

Comment:
This sounds reasonable, but can we use `RELEASE_ASSERT` for the scenarios where we might just see surprises from the environment, e.g. kernel? That way we don't silently ignore and just die. We could also just test and `NOT_IMPLEMENTED`. Both these are equivalent..